apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -169,7 +203,11 @@ var MozMobileMessageShim = {
     };
 
     cursor.onerror = function onerror() {
-      console.error('Reading the database. Error: ' + this.error.name);
+      debug(
+        'Reading the database. Error: %s',
+        this.error.message || this.error.name,
+        this.error
+      );
       stream.abort();
```

Comment:
I'll feed ```e.message || e.name``` as ```Error and Function objects cannot be duplicated by the structured clone algorithm; attempting to do so will throw a DATA_CLONE_ERR exception.``` [1]

However, since it's impossible to clone errors, should not bridge provide some help here? E.g. if  1) service method throws instance of Error or 2) returns rejected promise with such type or 3) stream is cancelled or 4) stream is aborted then bridge extracts as much error info as possible (message, name, type, line number, stack and etc), compose clonable plane object and then just reconstructs custom error object on the opposite side.

@wilsonpage how does that sound? Is it feasible?

[1] https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/The_structured_clone_algorithm#Things_that_don%27t_work_with_structured_clones